### PR TITLE
01.05.01 attendance milestones

### PR DESCRIPTION
**Pull request by ethanchung936**
- Added attendance milestone viewing
- Added progress updates in the form of notifications and confetti when new milestones are reached
- Cleaned up UI in a few spots to make milestones and attendees look pretty
- Moved notification sending to a different thread

If someone is reviewing, please test things out before approving, mainly focusing on the event side and attendees checking in.

### CODE REVIEW

File: code/noRAM/app/src/main/java/com/example/noram/AttendeeAnnouncementsActivity.java [hunk 1]
```diff
@@ -32,6 +42,31 @@ protected void onCreate(Bundle savedInstanceState){
 
         // retrieve corresponding event in database
         int eventID = getIntent().getIntExtra(eventIDLabel,0);
-        // TODO: Actually query database to get corresponding event
+
+        MainActivity.db.getEventsRef().document(String.valueOf(eventID)).get().addOnSuccessListener(documentSnapshot -> {
+            if (documentSnapshot.exists()) {
+                Event event = documentSnapshot.toObject(Event.class);
+                // displayAnnouncements(event);
```
Comment (ColeDewis) [inline code/noRAM/app/src/main/java/com/example/noram/AttendeeAnnouncementsActivity.java:49]: is this meant to be commented out

**Review by ColeDewis — COMMENTED** [review]
Will try to test this on my phone later today if I have time, left one small comment for now.

Also, to be clear, this also seems to add Christiaan's announcement stuff as well?

File: code/noRAM/app/src/main/java/com/example/noram/AttendeeAnnouncementsActivity.java (hunk 1 above)
Comment (ethanchung936) [reply code/noRAM/app/src/main/java/com/example/noram/AttendeeAnnouncementsActivity.java:49]: I'm not sure actually, this was from Chris' stuff. @WhoShotCupidd

Comment (WhoShotCupidd) [reply code/noRAM/app/src/main/java/com/example/noram/AttendeeAnnouncementsActivity.java:49]: yes it is but this stuff is slightly behind my branch so ignore it for now

File: code/noRAM/app/src/test/java/com/example/noram/MilestoneUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.example.noram.model.Milestone;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for the Milestone class.
```
Comment (WhoShotCupidd) [inline code/noRAM/app/src/test/java/com/example/noram/MilestoneUnitTest.java:16]: author pwez

**Review by WhoShotCupidd — APPROVED** [review]
besides that one comment the commenting and docs are great thx!

**Comment by ethanchung936** [comment]
> Will try to test this on my phone later today if I have time, left one small comment for now.
> 
> Also, to be clear, this also seems to add Christiaan's announcement stuff as well?

Yes, it has some of the notification stuff Christiaan was working on, but that part isn't fully finished

File: code/noRAM/app/src/test/java/com/example/noram/MilestoneUnitTest.java (hunk 2 above)
Comment (ethanchung936) [reply code/noRAM/app/src/test/java/com/example/noram/MilestoneUnitTest.java:16]: added it there and on all the other test files

**Pull request merged**